packages/date-picker/src/vaadin-date-picker-mixin.js

```diff
@@ -804,10 +851,13 @@ export const DatePickerMixin = (subclass) =>
         this.validate();
       }
 
-      // If the input isn't focused when overlay closes (fullscreen mode), clear focused state
-      if (this.getRootNode().activeElement !== this.inputElement) {
-        this._setFocused(false);
-      }
+      // Wait for focus to be restored on close by the `vaadin-overlay`.
+      setTimeout(() => {
+        // If the input isn't focused (fullscreen mode), clear focused state.
+        if (!isElementFocused(this._nativeInput)) {
+          this._setFocused(false);
+        }
+      });
```

Comment:
I wonder if this could be removed altogether? Couldn't reproduce a scenario where the component would have focused state on fullscreen mode.

Reply:
Yes, apparently this code can be removed. The only case when `focused` can be set is when `autoOpenDisabled` is also set, but in that case focus is correctly restored, so there is no need to reset this state anymore.